### DISCUSSION TITLE
How do i use my own/custom `chi_sim.traineddata`?

### DISCUSSION THREAD

**Discussion by piscopancer** [naptha/tesseract.js#908] [Q&A] [answered]
*Node 20, Next*

![image](https://github.com/naptha/tesseract.js/assets/109352196/9c778ef1-5e48-473e-a257-96d0086fdc55)

How do i use this traineddata? where should it go? the docs is too brief about using custom .traineddata files. Can i upload it remotely and pass a url or should i store it in my project and import it? I generally have very blurred understanding of what's going on in this config even after reading the docs
```ts
const worker = await createWorker('chi_sim', 1, {
  langPath: '', // ?
  dataPath: '', // ?
})
```

**Comment by Balearica** [comment, marked as answer]
For Node.js only (not browser), you can point Tesseract.js to a file on your local file system.  You would (1) set `langPath` to the directory that contains the `.traineddata` files and (2) set the `gzip` argument to `false` to tell Tesseract.js to look for `.traineddata` files rather than `.traineddata.gz` files.  

For example, if the path to one of your files is `./custom_data/chi_sim.traineddata`, you could set the following arguments.

```js
  const worker = await createWorker("chi_sim", 1, {
    langPath: './custom_data',
    gzip: false,
  });  
```

**Reply by piscopancer** [reply]
thank you

**Comment by Tieamone** [comment]
I need help because my script cannot invoke the chi_sim.traineddata. The error feedback is as follows:


F:\python\python.exe F:\pyt\pyt_caiji\test.py 
Current working directory: F:\pyt\pyt_caiji
Image file path: F:\pyt\pyt_caiji\test_image.png
Error occurred while capturing or processing text: (1, 'Error opening data file "C:\\Program Files\\Tesseract-OCR\\tessdata\\chi_sim.traineddata"/chi_sim.traineddata Please make sure the TESSDATA_PREFIX environment variable is set to your "tessdata" directory. Failed loading language \'chi_sim\' Tesseract couldn\'t load any languages! Could not initialize tesseract.')

Process finished with exit code 0
This error message indicates that Tesseract OCR is unable to locate or load the chi_sim.traineddata file for Chinese (Simplified) language support. It suggests checking if the TESSDATA_PREFIX environment variable is correctly set to the directory containing the tessdata folder. Additionally, it reports that Tesseract failed to load the specified language and could not initialize properly.

If you need assistance resolving this issue, please let me know!
